app/models/manageiq/providers/redfish/physical_infra_manager/operations/firmware.rb

```diff
@@ -0,0 +1,55 @@
+module ManageIQ::Providers::Redfish
+  module PhysicalInfraManager::Operations::Firmware
+    def update_firmware_async(firmware_binary, servers)
+      validate_update_firmware(firmware_binary, servers)
+
+      with_provider_connection do |client|
+        update_service = client.UpdateService
+        raise MiqException::MiqFirmwareUpdateError, 'UpdateService is not enabled' unless update_service
+
+        protocol, url = compatible_firmware_url(update_service, firmware_binary)
+
+        response = update_service.post(
+          :path    => update_service.Actions['#UpdateService.SimpleUpdate'].target,
+          :payload => {
+            :ImageURI         => url,
+            :TransferProtocol => protocol,
+            :Targets          => servers.map(&:ems_ref),
+          }
+        )
+
+        if response.status.to_i != 200
+          raise MiqException::MiqFirmwareUpdateError, "Cannot update firmware: (#{response.status}) #{response.data[:body]}"
+        end
+
+        response
+      end
+    end
+
+    def compatible_firmware_url(update_service, firmware_binary)
+      update_action = update_service.Actions['#UpdateService.SimpleUpdate']
+      requested = update_action['TransferProtocol@Redfish.AllowableValues']
+      requested ||= begin
+        info = update_action['TransferProtocol@Redfish.ActionInfo'].Parameters.find { |p| p.Name == 'TransferProtocol' }
+        info&.AllowableValues
+      end
+      raise MiqException::MiqFirmwareUpdateError, 'Redfish supports zero transfer protocols' if requested.empty?
```

Comment:
`requested` can be `nil` if both methods for obtaining supported protocols fail.